src/test/ui/typeck/issue-91450-inner-ty-error.rs

```diff
@@ -0,0 +1,7 @@
+// Regression test for #91450.
+// This test ensures that the compiler does not suggest `Foo<[type error]>` in diagnostic messages.
+
+fn foo() -> Option<_> {} //~ ERROR: [E0308]
+//~^ ERROR: the type placeholder `_` is not allowed
```

Comment:
I wanted to add a "negative" test of "help: replace with the correct return type: `Foo<[type error]>`" here to detect any regressions in the future but IIRC there isn't such annotation ;)